subprojects/docs/src/docs/userguide/organizingGradleProjects.adoc

```diff
@@ -15,55 +15,122 @@
+.
+├── build.gradle
+├── gradle
+│   ├── build-scan.gradle
+│   ├── ide.gradle
+│   └── publishing.gradle
+└── settings.gradle
 ----
 
-This means that you can just put your build source code in this directory and stick to the layout convention for a Java/Groovy project (see <<javalayout>>).
+Gradle doesn't take a stance on how to organize script plugins. However, a sensible default is to use the subdirectory `gradle`.
 
-If you need more flexibility, you can provide your own `build.gradle`. Gradle applies the default build script regardless of whether there is one specified. This means you only need to declare the extra things you need. Below is an example. Notice that this example does not need to declare a dependency on the Gradle API, as this is done by the default build script:
+[[sec:build_sources]]
+=== Use `buildSrc` project to abstract imperative logic
+
+Complex build logic is usually a good candidate for being encapsulated either as custom task or binary plugin. Custom task and plugin implementations should not live in the build script. It is very convenient to use `buildSrc` project for that purpose as long as the code is does not need to be shared among multiple, independent projects.
+
+The directory `buildSrc` is treated as a special project. Upon discovery of the directory, Gradle automatically compiles and tests this code and puts it in the classpath of your build script. For multi-project builds there can be only one `buildSrc` directory, which has to be in the root project directory.
+
+`buildSrc` uses the same <<javalayout,source code conventions>> applicable to Java and Groovy projects. It also provides direct access to the Gradle API. Additional dependencies can be declared in a dedicated `build.gradle` under `buildSrc`.
 
 ++++
 <sample xmlns:xi="http://www.w3.org/2001/XInclude" id="customBuildSrcBuild" dir="java/multiproject" title="Custom buildSrc build script">
     <sourcefile file="buildSrc/build.gradle"/>
 </sample>
 ++++
 
-The `buildSrc` project can be a multi-project build, just like any other regular multi-project build. However, all of the projects that should be on the classpath of the actual build must be `runtime` dependencies of the root project in `buildSrc`. You can do this by adding this to the configuration of each project you wish to export:
+A typical project including `buildSrc` has the following layout.
 
-++++
-<sample xmlns:xi="http://www.w3.org/2001/XInclude" id="multiProjectBuildSrc" dir="multiProjectBuildSrc" includeLocation="true" title="Adding subprojects to the root buildSrc project">
-    <sourcefile file="buildSrc/build.gradle" snippet="addToRootProject"/>
-</sample>
-++++
+----
+.
+├── build.gradle
+├── buildSrc
+│   └── src
+│       ├── main
+│       │   └── java
+│       │       ├── Deploy.java
+│       │       └── DeploymentPlugin.java
+│       └── test
+│           └── java
+│               └── DeploymentPluginTest.java
+├── client
```

Comment:
I think we can change client, service, and web into "src/main" or something. No need to tie this to multi-project builds and make users wonder what these directories are for.

Reply:
done. I actually didn't even use `src/main` here as it's also not really necessary.